### PR TITLE
DP-1 done

### PR DESCRIPTION
**Pull request by pradhan1234**
Please review @kaurramanpreet

### CODE REVIEW

File: HouseRobber.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        if (len == 2) {
+            return Math.max(nums[0], nums[1]);
+        }
+        // using dynamic programming to store solutions to overlapping subproblems.
+        int[] dp = new int[len];
+        dp[0] = nums[0];
+        dp[1] = Math.max(nums[0], nums[1]);
+        for (int i = 2; i < len; i++) {
+            // should we choose nums[i] depends on whether nums[i] + dp[i-2] 
+            // is optimal then dp[i-1]
```
Comment (kaurramanpreet) [inline HouseRobber.java]: can you elaborate your doubt
i didnt get it!!

Comment (pradhan1234) [reply HouseRobber.java]: Not doubt, it was a comment to jot down my reasoning let me rephrase it :)

Comment (pradhan1234) [reply HouseRobber.java]: "To Be or Not to be", 
that's the decision that drives dp, 

so will nums[i] be included (robbed) depends if nums[i] + dp[i-2] is more than dp[i], in which we skip robbing current house
